### PR TITLE
style(eslint-config): enable more rules to be consistent

### PR DESCRIPTION
**Pull request by merceyz**
**What's the problem this PR addresses?**

The eslint config is missing some "critical" rules to keep us consistent in the codebase.

Closes https://github.com/yarnpkg/berry/pull/2293

**How did you fix it?**

Enabled some more eslint rules, though even with the new set of rules it's not possible to keep the codebase consistent as eslint is for linting, not formatting, we need Prettier for that, but it's an improvement.

**Checklist**
- [x] I have read the [Contributing Guide](https://yarnpkg.com/advanced/contributing).
- [x] I have set the packages that need to be released for my changes to be effective.
- [x] I will check that all automated PR checks pass before the PR gets reviewed.

### CODE REVIEW

File: packages/eslint-config/rules/style.js [hunk 1]
```diff
@@ -50,8 +50,12 @@ module.exports = {
 
     'arrow-spacing': 2,
 
+    'brace-style': 2,
+
     'comma-dangle': [`error`, `always-multiline`],
 
+    'comma-spacing': 2,
```
Comment (paul-soporan) [inline packages/eslint-config/rules/style.js:57]: You should enable `@typescript-eslint/comma-spacing`, just like I did in my PR. :P

File: packages/eslint-config/rules/style.js [hunk 2]
```diff
@@ -50,8 +50,12 @@ module.exports = {
 
     'arrow-spacing': 2,
 
+    'brace-style': 2,
```
Comment (paul-soporan) [inline packages/eslint-config/rules/style.js:53]: Same thing with `brace-style`.

File: packages/eslint-config/rules/style.js (hunk 1 above)
Comment (merceyz) [reply packages/eslint-config/rules/style.js:57]: Yup, noticed, going over more rules atm

File: packages/eslint-config/rules/style.js [hunk 3]
```diff
@@ -9,6 +9,14 @@ module.exports = {
       default: `generic`,
     }],
 
+    '@typescript-eslint/brace-style': 2,
+
+    '@typescript-eslint/comma-dangle': [`error`, `always-multiline`],
+
+    '@typescript-eslint/comma-spacing': 2,
+
```
Comment (merceyz) [inline packages/eslint-config/rules/style.js:17]:
```suggestion
```

**Pull request merged**